Tella/Utils/NetworkMonitor.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Network
+import Combine
+
+class NetworkMonitor: ObservableObject {
+    
+    var connexionDidChange = PassthroughSubject<Bool,Never>()
+    var isConnected : Bool = true
+    
+    private let monitor = NWPathMonitor()
+    private let queue = DispatchQueue(label: "Monitor")
+    
+    static let shared = NetworkMonitor()
+    
+    init() {
+        monitor.pathUpdateHandler = { path in
+            DispatchQueue.main.async {
+                let newStatus =  path.status == .satisfied
+                self.connexionDidChange.send(newStatus)
+                self.isConnected = newStatus
+            }
+        }
+        monitor.start(queue: queue)
+    }
```

Comment:
Maybe we should add a clean up method? to ensure to stop the NWPathMonitor when it's no longer needed. Let's add add deinit method to stop the monitor and remove the observer.